src/App.ts

```diff
@@ -312,9 +316,33 @@ export default class App {
     );
   }
 
-  public view(callbackId: string | RegExp, ...listeners: Middleware<SlackViewMiddlewareArgs>[]): void {
+  public view(
+    callbackId: string | RegExp,
+    ...listeners: Middleware<SlackViewMiddlewareArgs>[]
+  ): void;
+  public view(
+    constraints: ViewConstraints,
+    ...listeners: Middleware<SlackViewMiddlewareArgs>[]
+  ): void;
+  public view(
+    callbackIdOrConstraints: string | RegExp | ViewConstraints,
+    ...listeners: Middleware<SlackViewMiddlewareArgs>[]): void {
+    const constraints: ViewConstraints =
+      (typeof callbackIdOrConstraints === 'string' || util.types.isRegExp(callbackIdOrConstraints)) ?
+      { callback_id: callbackIdOrConstraints, type: 'view_submission' } : callbackIdOrConstraints;
+
+      // Fail early if the constraints contain invalid keys
+    const unknownConstraintKeys = Object.keys(constraints)
+      .filter(k => (k !== 'callback_id' && k !== 'type'));
```

Comment:
are we 💯% satisfied with using the constraint name `type`? i feel like its a little too generic.

Reply:
if we're doing validations on the constraints, then does it make sense to check the value of `type` is one of the two acceptable values? in other methods, we don't because the values are just strings, so there's nothing specific we can validate.

Reply:
So far constraints have aligned with the actual payload that the developer receives. I'm fine not using `type`, but it could be confusing and would not align with the platform's description of the event.

But yes on the validation of whether it's one of the two strings. I'll add that change.